src/main/java/ua/lviv/iot/searcher/Searcher.java

```diff
@@ -0,0 +1,35 @@
+package ua.lviv.iot.searcher;
+
+import java.util.ArrayList;
+import java.util.List;
+
+import ua.lviv.iot.graph.Graph;
+
+public class Searcher {
+
+	public List<String> findOptimalSequenceFor(Graph<String> graph) {
+		int numberOfVertices = graph.numberOfVertices();
+		List<String> optimalSequence = new ArrayList<String>(numberOfVertices);
+		boolean[] isVisited = new boolean[numberOfVertices];
+		for (int counter = 0; counter < numberOfVertices; counter++) {
+			isVisited[counter] = false;
```

Comment:
Know your language! Learn the bits and tricks of it. Why do you think I post this comment here?